include/nil/crypto3/zk/snark/systems/plonk/placeholder/lookup_argument.hpp

```diff
@@ -51,291 +51,448 @@ namespace nil {
-                                            value_assignment = plonk_columns.witness(
-                                                lookup_gates[i].constraints[j].lookup_value[k].index);
-                                            break;
-                                        case VariableType::column_type::public_input:
-                                            value_assignment = plonk_columns.public_input(
-                                                lookup_gates[i].constraints[j].lookup_value[k].index);
-                                            break;
-                                        case VariableType::column_type::constant:
-                                            value_assignment = plonk_columns.constant(
-                                                lookup_gates[i].constraints[j].lookup_value[k].index);
-                                            break;
-                                        case VariableType::column_type::selector:
-                                            break;
-                                    }
-                                    for (std::size_t t = 0; t < basic_domain->m; t++) {
-                                        F_compr_input[t] =
-                                            F_compr_input[t] +
-                                            theta_acc *
-                                                input_assignment[(j + lookup.get_vars()[0].rotation) %
-                                                                 input_assignment.size()] *
-                                                lookup.get_coeff() *
-                                                plonk_columns.selector(lookup_gates[i].selector_index)[t];
-                                        F_compr_value[t] =
-                                            F_compr_value[t] +
-                                            theta_acc * value_assignment[j] *
-                                                plonk_columns.selector(lookup_gates[i].selector_index)[t];
+                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> sorted(reduced_input.size() + reduced_value.size());
+                        for( std::size_t i = 0; i < sorted.size(); i++ ){
+                            sorted[i] = math::polynomial_dfs<typename FieldType::value_type>(basic_domain->m-1, basic_domain->m, FieldType::value_type::zero());
+                        }
+                        std::size_t i1=0;
+                        std::size_t j1=0;
+                        typename FieldType::value_type prev(0);
+                        prev = typename FieldType::value_type(0);
+                        for( std::size_t i = 0; i < reduced_value.size(); i++){
+                            for( std::size_t j = 0; j < preprocessed_data.common_data.usable_rows_amount; j++){
+                                if(reduced_value[i][j] != prev){
```

Comment:
I do NOT understand this algorithm, but without trying to learn the lookup argument algorithm, I have a feeling that this code is wrong. But maybe it's ok.

I assume you check here that the value is not repeating, but these values are not sorted right? Can't it repeat sometime later? Let's say I have 10 copies for some value "prev", but they are not consecutive? Then I will place 10*10 values into the variable "sorted" no?

Again, maybe this is ok, I don't understand the algorithm itself.

Reply:
It's ok for our parameters and lookup table construction algorithm. Probability of collision is negligible.